### ISSUE TITLE
Add a script event for dump-autoload command

### ISSUE THREAD

**Issue by xrstf** [composer/composer#1447]
I'm trying to create a [package](https://bitbucket.org/xrstf/composer-php52) providing a script command to create a PHP 5.2 compatible autoloader in parallel to the original, PHP 5.3 autoloader.

This works well for install and update commands, but I have currently no chance to update the autoloader when someone uses the `dump-autoload` command. This is especially painful as I think a lot of people will use this command to generate the optimized, classmap-based autoloader (instead of passing `-o` to the install/update command).

I can see other custom autoloading stuff happening in other projects, so it would be nice to have a `post-autoload-cmd` event.
